integration-tests/src/test/java/oracle/weblogic/kubernetes/ItClusterResourceScaling.java

```diff
@@ -0,0 +1,527 @@
+          .append(decodedToken)
+          .append("\" ");
+    }
+    command.append("-H Accept:application/json ")
+        .append("-H Content-Type:application/json ");
+    if (hasHeader) {
+      command.append("-H X-Requested-By:MyClient ");
+    }
+    command.append("-d '{\"spec\": {\"replicas\": ")
+    .append(numOfServers)
+    .append("}}' ")
+    .append("-X POST https://")
+    .append(getHostAndPort(opExternalSvc, externalRestHttpsPort))
+    .append("/operator/latest/domains/")
+    .append(domainUid)
+    .append("/clusters/")
+    .append(clusterName)
+    .append("/scale").toString();
+
+    CommandParams params = Command
+        .defaultCommandParams()
+        .command(command.toString())
+        .saveResults(true)
+        .redirect(true);
+
+    logger.info("Calling curl to scale the cluster");
+    ExecResult result = Command.withParams(params).executeAndReturnResult();
+    logger.info("Return values {0}, errors {1}", result.stdout(), result.stderr());
+    if (result != null) {
+      logger.info("Return values {0}, errors {1}", result.stdout(), result.stderr());
+      if (result.stdout().contains(expectedMsg) || result.stderr().contains(expectedMsg)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+
+  private String getServerToken(String opServiceAccount) {
+    logger.info("Getting the secret of service account {0} in namespace {1}", opServiceAccount, opNamespace);
+    String secretName = Secret.getSecretOfServiceAccount(opNamespace, opServiceAccount);
+    if (secretName.isEmpty()) {
+      logger.info("Did not find secret of service account {0} in namespace {1}", opServiceAccount, opNamespace);
+      return null;
+    }
+    logger.info("Got secret {0} of service account {1} in namespace {2}",
+        secretName, opServiceAccount, opNamespace);
+
+    logger.info("Getting service account token stored in secret {0} to authenticate as service account {1}"
+        + " in namespace {2}", secretName, opServiceAccount, opNamespace);
+    String secretToken = Secret.getSecretEncodedToken(opNamespace, secretName);
+    if (secretToken.isEmpty()) {
+      logger.info("Did not get encoded token for secret {0} associated with service account {1} in namespace {2}",
+          secretName, opServiceAccount, opNamespace);
+      return null;
+    }
+    logger.info("Got encoded token for secret {0} associated with service account {1} in namespace {2}: {3}",
+        secretName, opServiceAccount, opNamespace, secretToken);
+    return secretToken;
+  }
+
+  private boolean createVerifyDomain(String domainNamespace, String domainUid) {
+
+    // create and verify the domain
+    logger.info("Creating and verifying model in image domain");
+    createAndVerifyMiiDomain(domainNamespace, domainUid);
+    return true;
+  }
+
+  /**
+   * Create a model in image domain and verify the domain pods are ready.
+   */
+  private void createAndVerifyMiiDomain(String domainNamespace, String domainUid) {
```

Comment:
The most of parts of this method is similar to CommonMiiTestUtils#createMiiDomainAndVerify. You can check whether you can reuse that method.

Reply:
done